infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/SszCollectionSchema.java

```diff
@@ -44,20 +44,19 @@ default SszCollectionT of(final SszElementT... elements) {
     return createFromElements(List.of(elements));
   }
 
-  @SuppressWarnings("unchecked")
   default SszCollectionT createFromElements(final List<? extends SszElementT> elements) {
+    return createFromBackingNode(createTreeFromElements(elements));
+  }
+
+  default TreeNode createTreeFromElements(final List<? extends SszElementT> elements) {
+    // TODO: suboptimal
```

Comment:
What's subpotimal? creating a mutable entity and filling it by elements? I'd at least move this comment a line below, so it would be easy to guess. Or add some clarity

Reply:
Yep, added explanations. The implementation is probably not that bad in terms of performance. So I'm leaving it as is until profiling finds a bottleneck here